compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3579,74 +3629,11 @@ abstract class GenJSCode extends plugins.PluginComponent
 
     /** Gen JS code for a coercion */
     private def genCoercion(tree: Apply, receiver: Tree, code: Int): js.Tree = {
```

Comment:
Ah, somehow I interpreted that you wanted the first overload of `genConversion` to be inlined, not this one. I prefer this one left as a separate method, because its call site is a series of `if (isXYZOp(code)) genXYZ(tree)`. I think it makes sense not to break that pattern.